src/states/elder/mod.rs

```diff
@@ -2083,14 +2084,25 @@ impl Approved for Elder {
 
         self.update_peer_states(outbox);
 
-        if self_sec_update {
+        let dst = if self_sec_update {
             self.chain.reset_candidate_if_member_of(sec_info.members());
             self.send_neighbour_infos();
+
+            Authority::Section(*self.name())
         } else {
             self.remove_from_proving_section_cache(&sec_info);
             // Vote for neighbour update if we haven't done so already.
             // vote_for_event is expected to only generate a new vote if required.
-            self.vote_for_event(sec_info.into_network_event());
+            self.vote_for_event(sec_info.clone().into_network_event());
+
+            Authority::Section(sec_info.prefix().name())
+        };
+
+        if !is_initial {
```

Comment:
Do we really need this `is_initial`?
When it is true it seem to be for our own section?